presto-hudi/src/main/java/com/facebook/presto/hudi/split/HudiPartitionSplitGenerator.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.hudi.split;
+
+import com.facebook.airlift.log.Logger;
+import com.facebook.presto.hive.metastore.ExtendedHiveMetastore;
+import com.facebook.presto.hive.metastore.MetastoreContext;
+import com.facebook.presto.hive.util.AsyncQueue;
+import com.facebook.presto.hudi.HudiFile;
+import com.facebook.presto.hudi.HudiPartition;
+import com.facebook.presto.hudi.HudiSplit;
+import com.facebook.presto.hudi.HudiTableHandle;
+import com.facebook.presto.hudi.HudiTableLayoutHandle;
+import com.facebook.presto.hudi.HudiTableType;
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.ConnectorSplit;
+import com.facebook.presto.spi.schedule.NodeSelectionStrategy;
+import com.google.common.collect.ImmutableList;
+import io.airlift.units.DataSize;
+import org.apache.hadoop.fs.Path;
+import org.apache.hudi.common.fs.FSUtils;
+import org.apache.hudi.common.model.FileSlice;
+import org.apache.hudi.common.table.view.HoodieTableFileSystemView;
+import org.apache.hudi.common.util.HoodieTimer;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Queue;
+
+import static com.facebook.presto.hudi.HudiMetadata.toMetastoreContext;
+import static com.facebook.presto.hudi.HudiSessionProperties.getMinimumAssignedSplitWeight;
+import static com.facebook.presto.hudi.HudiSessionProperties.getStandardSplitWeightSize;
+import static com.facebook.presto.hudi.HudiSessionProperties.isSizeBasedSplitWeightsEnabled;
+import static com.facebook.presto.hudi.HudiSplitManager.getHudiPartition;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * A runnable to take partition names from a queue of partitions to process,
+ * generate Hudi splits from the partition, and add the splits to the async
+ * ConnectorSplit queue.
+ */
+public class HudiPartitionSplitGenerator
+        implements Runnable
+{
+    private static final Logger log = Logger.get(HudiPartitionSplitGenerator.class);
+
+    private final ExtendedHiveMetastore metastore;
+    private final MetastoreContext metastoreContext;
+    private final HudiTableLayoutHandle layout;
+    private final HudiTableHandle table;
+    private final Path tablePath;
+    private final HoodieTableFileSystemView fsView;
+    private final AsyncQueue<ConnectorSplit> asyncQueue;
+    private final Queue<String> concurrentPartitionQueue;
```

Comment:
How do we handle non partitioned tables?

Reply:
For non-partitioned tables just 1 partition i.e. empty partition is added to the queue and only one thread will create splits, whoever picks from the queue. For the rest of the threads, it will be no-op.